server/stream.go

```diff
@@ -876,14 +854,23 @@ func (mset *stream) setLeader(isLeader bool) error {
 		if err := mset.subscribeToStream(); err != nil {
 			if mset.isClustered() {
 				// Stepdown since we have an error.
-				mset.stepDown()
 				mset.node.StepDown()
 			}
 			mset.mu.Unlock()
 			return err
 		}
 	} else {
-		mset.stepDown()
+		// cancel timer to create the source consumers if not fired yet
+		if mset.sourcesConsumerSetup != nil {
+			mset.sourcesConsumerSetup.Stop()
+			mset.sourcesConsumerSetup = nil
+		}
```

Comment:
This else clause got dropped from above `mset.stopSourceConsumers()`

Reply:
The else clause was part of the cleanup that I had initially added and that you asked me to remove

Reply:
Let's add that one in, but put conditional so code reads a bit cleaner, so `if ... {`

Reply:
ok done.